controllers/loadtest_controller.go

```diff
@@ -73,7 +72,29 @@ func (r *LoadTestReconciler) Reconcile(req ctrl.Request) (ctrl.Result, error) {
 		return ctrl.Result{}, client.IgnoreNotFound(err)
 	}
 
+	loadTestTTL := time.Duration(rawTest.Spec.TTLSeconds) * time.Second
+	loadTestTimeout := time.Duration(rawTest.Spec.TimeoutSeconds) * time.Second
+
+	if loadTestTTL == 0 {
+		log.Info("TTL is invalid", "TTL", loadTestTTL)
+	}
+
+	if loadTestTimeout == 0 {
+		log.Info("Timeout is invalid", "Timeout", loadTestTimeout)
+	}
+
+	if loadTestTimeout > loadTestTTL {
+		log.Info("loadTestTTL is less than loadTestTimeout", "Timeout", loadTestTimeout, "TTL", loadTestTTL)
+	}
+
 	if rawTest.Status.State.IsTerminated() {
+		if time.Now().Sub(rawTest.Status.StartTime.Time) >= loadTestTTL {
+			log.Info("test expired, deleting", "startTime", rawTest.Status.StartTime, "loadTestTTL", loadTestTTL)
+			if err = r.Delete(ctx, rawTest); err != nil {
+				log.Error(err, "fail to delete test", "name", req.NamespacedName)
```

Comment:
We can leave the name of the load test off, because it is automatically added to every log statement in reconcile. 

```suggestion
				log.Error(err, "fail to delete test")
```